### DISCUSSION TITLE
Multiplayer: Add ban IP feature / action log for players / setting password during game

### DISCUSSION THREAD

**Discussion by atmaxinger** [OpenRCT2/OpenRCT2#15662] [Feature requests]
As there currently are some really nice people in multiplayer who seem to have nothing better to do than to destroy other peoples parks it would be nice to be able to not only kick players but also ban them (e.g. using the IP of the player or some kind of unique system hash).

Alternatively setting a password in-game could fix this.

Also some kind of record of what player did what (e.g. delete rollercoaster, remove footpath, ...) would be great.

**Comment by janisozaur** [comment]
Banning players was already discussed and is not something we're going to work on. Feel free to contribute that though.

Setting a password is already an option, but cannot be changed mid-game, if that's what you mean.

**Comment by atmaxinger** [comment]
Yeah, I mean mid-game. 

I'll take a look at banning when I've got some time on my hands.

**Comment by janisozaur** [comment]
https://github.com/OpenRCT2/OpenRCT2/issues/2351

**Comment by janisozaur** [comment]
This will be fixed with #3699

**Comment by Nubbie** [comment]
Add ban IP feature: **De-voted**
action log for players: **Implemented**
setting password during game: _(unknown)_ (closed PR #2769)

**Comment by janisozaur** [comment]
If there's another issues for setting a password while the server is already started, then this can be closed. If not, this issue should be changed to a feature request for changing said password.
